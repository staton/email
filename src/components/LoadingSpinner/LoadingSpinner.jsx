import React, { Component } from 'react';

class LoadingSpinner extends Component {

	render() {
        // Google Loader source: https://codepen.io/jczimm/pen/vEBpoL
        return (
            <div className="LoadingSpinner">
                <div className="LoadingSpinner__loader">
                    <svg 
                        className="LoadingSpinner__circular" 
                        viewBox="25 25 50 50">
                    <circle 
                        className="LoadingSpinner__path" 
                        cx="50" 
                        cy="50" 
                        r="20" 
                        fill="none" 
                        strokeWidth="4" 
                        strokeMiterlimit="10"/>
                    </svg>
                </div>
            </div>
        );
    }

}

export default LoadingSpinner;