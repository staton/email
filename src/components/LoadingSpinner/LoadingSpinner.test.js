import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import LoadingSpinner from './LoadingSpinner';

describe('<LoadingSpinner />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<LoadingSpinner />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    
});