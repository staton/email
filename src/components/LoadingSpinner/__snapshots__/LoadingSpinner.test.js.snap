// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> renders correctly 1`] = `
<div
  className="LoadingSpinner"
>
  <div
    className="LoadingSpinner__loader"
  >
    <svg
      className="LoadingSpinner__circular"
      viewBox="25 25 50 50"
    >
      <circle
        className="LoadingSpinner__path"
        cx="50"
        cy="50"
        fill="none"
        r="20"
        strokeMiterlimit="10"
        strokeWidth="4"
      />
    </svg>
  </div>
</div>
`;
